test/cpp/jit/test_gpu.cpp

```diff
@@ -4605,6 +4605,330 @@ void testGPU_FusionReductionScheduler() {
       aten_output.sub(cg_output).abs().max());
 }
 
+void testGPU_FusionCacheBefore() {
+  // TVM Cache Write
+  torch::jit::fuser::cuda::CudaKernel prog;
+  prog.setFusionPtr(std::make_unique<Fusion>());
+  Fusion* fusion = prog.fusion();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeDummyTensor(2);
+  TensorView* tv1 = add(tv0, new Float(1.0));
+  TensorView* tv2 = mul(tv1, new Float(3.0));
+  fusion->addInput(tv0);
+  fusion->addOutput(tv2);
+  // Before: TV1 = TV0 + 1
+  // After: TV1 = TV2; TV2 = TV0 + 1
```

Comment:
Are these comments correct?

Reply:
I changed the test but forgot to update the comments.

Reply:
` // Before: TV2 = TV1 * 3
  // After:  TV3 = TV1 * 3;
  //         TV2 = TV3;
  // Algorithm`

Reply:
Looks like there are also other inconsistent comments. Can you go through the other tests as well? It would be very helpful to understand what is really intended to do.

Reply:
Example: https://github.com/csarofeen/pytorch/pull/166/files#diff-538e381bc787011cc53c15caf54e126dR4678

Reply:
The comments for `testGPU_FusionCacheAfter` should be correct. I just reviewed them.

Reply:
Ah, I see. I was just wondering why it didn't have `tv2`.